include/nwgraph/adjacency.hpp

```diff
@@ -85,7 +85,7 @@ class index_adjacency : public unipartite_graph_base, public indexed_struct_of_a
    * @brief Constructor of index_adjacency. Require the type of the graph to be unipartite.
    * Create an empty index_adjacency.
    */
-  index_adjacency(std::array<size_t, 1> N, size_t M = 0) requires(std::is_same<unipartite_graph_base, unipartite_graph_base>::value) : unipartite_graph_base(N), base(N[0], M) {}
+  index_adjacency(std::array<size_t, 1> N, size_t M = 0) requires(std::is_same<unipartite_graph_base, unipartite_graph_base>::value) : unipartite_graph_base(N[0]), base(N[0], M) {}
```

Comment:
Hi @lums658 , should `unipartite_graph_base` be accepting `N[0]` here instead of an `array`?  I don't see a constructor that accepts a range, only an integer.